test/e2e/daemon_set.go

```diff
@@ -374,3 +422,24 @@ func checkRunningOnAllNodes(f *framework.Framework, selector map[string]string)
 func checkRunningOnNoNodes(f *framework.Framework, selector map[string]string) func() (bool, error) {
 	return checkDaemonPodOnNodes(f, selector, make([]string, 0))
 }
+
+func checkDaemonPodsImage(f *framework.Framework, selector map[string]string, image string) func() (bool, error) {
+	return func() (bool, error) {
+		selector := labels.Set(selector).AsSelector()
+		options := api.ListOptions{LabelSelector: selector}
+		podList, err := f.Client.Pods(f.Namespace.Name).List(options)
+		if err != nil {
+			return false, nil
+		}
+		pods := podList.Items
+
+		for _, pod := range pods {
+			podImage := pod.Spec.Containers[0].Image
+			if podImage != image {
+				err := fmt.Errorf("Wrong image for pod: %s. Expected: %s, got: %s", pod.Name, image, podImage)
+				return false, err
```

Comment:
We shouldn't return error here. It gives up the wait loop immediately. We should instead return `nil` to wait for the image to match.